Add a page with a description for services providing gateways (#301)
owncloud/docs-ocis: modules/ROOT/pages/deployment/services/gateways.adoc

= Gateways
:toc: macro

> :description: There is more than one service providing gateway functionalities specific to particular tasks beside other tasks these services handle. The following explanation helps sorting things out.

Author: ```suggestion
:description: There is more than one service providing gateway functionalities. The following explanation helps sorting things out.
```

> :description: There is more than one service providing gateway functionalities specific for particular tasks beside other tasks these services handle. The following explanation helps sorting things out.

Reviewer: ```suggestion
:description: There is more than one service providing gateway functionalities specific to particular tasks beside other tasks these services handle. The following explanation helps sorting things out.
```

include::partial$deployment/services/beta-statement.adoc[]

// remove the remark to get the toc rendered if there are more than one headlines
// {empty} +

// toc::[]

== Introduction

{description} The listing below focuses on the gateway and related functionalities of the mentioned services only.

== Services Providing Gateways

Proxy::

> The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.

Author: ```suggestion
The xref:{s-path}/proxy.adoc[proxy service] is a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
```

> The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages http related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming http requests to other services.

Reviewer: ```suggestion
The xref:{s-path}/proxy.adoc[proxy service] provides a general purpose HTTP API gateway and manages HTTP-related tasks like transport encryption, authentication and others. Note that the proxy service routes incoming HTTP requests to other services.
```

+

> NOTE: The proxy service should be for security reasons the only service that is accessible from the outside.

Reviewer: ```suggestion
NOTE: For security reasons, the proxy service should be the only service that is accessible from the outside.
```

> NOTE: For security reasons, the proxy service should be the only service that is accessible from the outside.

Author: ```suggestion
NOTE: All services end users currently can use, are provided by the proxy. All other APIs and services need to be protected to not be accessible by end users and are only for service-to-service communication.
```

Frontend::

> The xref:{s-path}/frontend.adoc[frontend service] is a datagateway and provides a http up- and download API. It for example connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.

Reviewer: ```suggestion
The xref:{s-path}/frontend.adoc[frontend service] is a data gateway and provides a HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
```

> The xref:{s-path}/frontend.adoc[frontend service] is a data gateway and provides a HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.

Author: ```suggestion
The xref:{s-path}/frontend.adoc[frontend service] the fronend acts among other things as the data gateway for the HTTP up- and download API. For example, it connects to the xref:{s-path}/storage-users.adoc[storage-users service] to manage up- and download activities based on requests via the proxy service.
```

Gateway::

> The xref:{s-path}/gateway.adoc[gateway service] provides a CS3 API gateway for all gRPC requests.

Author: ```suggestion
The xref:{s-path}/gateway.adoc[gateway service] provides a CS3 API gateway for gRPC requests to the CS3 backend.
```
